pkg/scaler/machineSetScaler.go

```diff
@@ -200,6 +200,8 @@ func (s *machineSetScaler) EnsureScaleDownNodes(c client.Client, nds drain.NodeD
 	err = c.List(context.TODO(), originalMachineSets, []client.ListOption{
 		client.InNamespace(MACHINE_API_NAMESPACE),
 		NotMatchingLabels{LABEL_UPGRADE: "true"},
+		client.MatchingLabels{"hive.openshift.io/machine-pool": "worker"},
+		client.MatchingLabels{"hive.openshift.io/machine-pool": "infra"},
```

Comment:
I'm not really sure how this would address the problem, having multiple label selectors match the same label in the `List` is only going to end up with one of them succeeding, not both.

ie. what this call is doing is equivalent to
`oc get machinesets -n openshift-machine-api -l hive.openshift.io/machine-pool=worker -l hive.openshift.io/machine-pool=infra`
which is only going to return the infra machineset as it was the last label selector in the list (reverse them, and it'll be the worker returned)